### ISSUE TITLE
Issue with xml.java package

### ISSUE THREAD

**Issue by raviteja713** [stleary/JSON-java#394]
Trying to convert JSON data into XML string using the below logic of code snippet. But, output xml has a JSONObject being displayed after "content" attribute as show below. Any suggestions.?

# **_# input.json:_**

```
{
  "id": "0001",
  "type": "donut",
  "name": "Cake",
  "payload": {
	"uname":"David"
    "content": {
      "batter": [
        {
          "id": "1001",
          "type": "Regular",
          "mode": [
            "rel",
            "don"
          ]
        }
      ]
    }
  }
}

```
# **_jsontoxml.java:_**

```
org.json.JSONObject jsonObject = new org.json.JSONObject(jsonData);
string xml="<?xml version=\"1.0\" encoding=\"ISO-8859-15\"?>\n<"+root+">" + org.json.XML.toString(jsonObject) + "</"+root+">
```

# _output.xml:_

```
<?xml version="1.0" encoding="ISO-8859-15"?>
<root>
    <id>0001</id>
    <type>donut</type>
    <name>Cake</name>  
	<payload>
        <uname>David</uname>
        {"batter": [{"id": "1001","type": "Regular","mode": ["rel","don"]}]}
    </payload>
</root>
```

**Comment by stleary** [comment]
"content" is an unfortunately-named keyword in the XML <-> Java transformation. For the history of this issue, please see #344, #286, and #108. For more information about how content is used, see XMLTest.java contentOperations() in https://github.com/stleary/JSON-Java-unit-test. 

No objection if someone wants to propose a workaround along the lines of #108, or any other approach that does not break existing applications.

**Comment by johnjaylward** [comment]
#108 looks like a decent implementation to me. I don't have time to work on it now, but may in a few weeks.

**Comment by johnjaylward** [comment]
@stleary what do you think of having a spplemental configuration object for the XML parsing. something like this:

```java
public class XMLParserConfiguration {
    /**
     * When parsing the XML into JSON, specifies if values should be kept as strings (true), or if
     * they should try to be guessed into JSON values (numeric, boolean, string)
     */
    public final boolean keepStrings;
    /**
     * The name of the key in a JSON Ojbect that indicates a CDATA section. Historically this has
     * been the value "content" but can be changed. Use <code>null</code> to indicate no CDATA
     * processing.
     */
    public final String cDataTagName;

    /**
     * Default parser configuration. Does not keep strings, and the CDATA Tag Name is "content".
     */
    public XMLParserConfiguration () {
          this(false, "content");
    }

    /**
     * Configure the parser string processing and use the default CDATA Tag Name as "content".
     * @param keepStrings <code>true</code> to parse all values as string.
     *      <code>false</code> to try and convert XML string values into a JSON value.
     */
    public XMLParserConfiguration (final boolean keepStrings) {
          this(keepStrings, "content");
    }

    /**
     * Configure the parser string processing to try and convert XML values to JSON values and
     * use the passed CDATA Tag Name the processing value. Pass <code>null</code> to
     * disable CDATA processing
     * @param cDataTagName<code>null</code> to disable CDATA processing. Any other value
     *      to use that value as the JSONObject key name to process as CDATA.
     */
    public XMLParserConfiguration (final String cDataTagName) {
          this(false, cDataTagName);
    }

    /**
     * Configure the parser string processing and use the default CDATA Tag Name as "content".
     * @param keepStrings <code>true</code> to parse all values as string.
     *      <code>false</code> to try and convert XML string values into a JSON value.
     * @param cDataTagName<code>null</code> to disable CDATA processing. Any other value
     *      to use that value as the JSONObject key name to process as CDATA.
     */
    public XMLParserConfiguration (final boolean keepStrings, final String cDataTagName) {
          this.keepStrings = keepStrings;
          this.cDataTagName = cDataTagName;
    }

}
```

The existing public functions would stay the same, but 1 new method could be added in to take the configuration

XML class:
```java
// keep this
public static JSONObject toJSONObject(String string){ ... }
// keep this
public static JSONObject toJSONObject(String string, boolean keepStrings){ ... }
// new
public static JSONObject toJSONObject(String string, XMLParserConfiguration  config){ ... }


private static boolean parse(XMLTokener x, JSONObject context, String name, XMLParserConfiguration  config) {
}

**Comment by stleary** [comment]
Agreed, looks like a reasonable way to address the issue.

**Comment by LukeButtersFunnelback** [comment]
Would it make sense to add support for supplying a key mapper so that we can re-map keys from json say "content" or "somethingElseXmlDoesn'tLike" into keys that XML would be happy with?

**Comment by stleary** [comment]
@LukeButtersFunnelback 'content' was Douglas' choice when first implementing this library. It has no special meaning as far as I am aware. It will not be changed under any circumstances due to a longstanding requirement not to change the behavior of existing code. The recommended solution by @johnjaylward still looks like the best way to proceed.

**Comment by LukeButtersFunnelback** [comment]
ah cool, I thought maybe "content" meant something special in JSON or XML.

**Comment by stleary** [comment]
Fixed in #412
